enclave/enclave-api/src/main/java/com/quorum/tessera/enclave/EncodedPayloadCodec.java

```diff
@@ -0,0 +1,32 @@
+package com.quorum.tessera.enclave;
+
+import com.quorum.tessera.version.BaseVersion;
+import com.quorum.tessera.version.CBORSupportVersion;
+import java.util.Set;
+import java.util.stream.Stream;
+
+public enum EncodedPayloadCodec {
+  CBOR(CBORSupportVersion.API_VERSION_5),
+  LEGACY(BaseVersion.API_VERSION_1);
+
+  String minimumSupportedVersion;
+
+  EncodedPayloadCodec(String minimumSupportedVersion) {
+    this.minimumSupportedVersion = minimumSupportedVersion;
+  }
+
+  public String getMinimumSupportedVersion() {
+    return minimumSupportedVersion;
+  }
+
+  public static EncodedPayloadCodec current() {
+    return CBOR;
+  }
+
+  public static EncodedPayloadCodec getPreferredCodec(Set<String> versions) {
+    return Stream.of(EncodedPayloadCodec.values())
+        .filter(codec -> versions.contains(codec.getMinimumSupportedVersion()))
+        .findFirst()
```

Comment:
The order of enum values() method is guaranteed - being the order they are declared. However it's true that this might not be the best practice - I'm adding a sort in case this declaration is changed in the future.